### ISSUE TITLE
How to use socks5 or http proxy to access telegram api?

### ISSUE THREAD

**Issue by benjamin-2019** [gram-js/gramjs#128]
Thanks to the author for providing this library and giving us a good development experience. This is the conclusion we have reached after trying all related JS libraries. However, the small plane has many limitations. A single account cannot frequently operate the API. I want to know how to use the proxy Access, as described in the python library: https://docs.telethon.dev/en/latest/basic/signing-in.html?highlight=proxy#signing-in-behind-a-proxy

**Comment by painor** [comment]
why do you actually want to use proxies? telegram has no IP limit

**Comment by benjamin-2019** [comment]
> why do you actually want to use proxies? telegram has no IP limit

Sir, there is no limit for a single account and a single client. When we need to manage hundreds or thousands of accounts on a computer at the same time, this will be limited. It needs dozens or hundreds of proxy IP support. We Now we provide customers with automatic group management services. We can only manage dozens of accounts on a computer. Without proxy IP, we can only add computers and purchase VPNs. That kind of cost is very expensive, and our current income cannot afford it.

**Comment by painor** [comment]
> > why do you actually want to use proxies? telegram has no IP limit
> 
> Sir, there is no limit for a single account and a single client. When we need to manage hundreds or thousands of accounts on a computer at the same time, this will be limited. It needs dozens or hundreds of proxy IP support. We Now we provide customers with automatic group management services. We can only manage dozens of accounts on a computer. Without proxy IP, we can only add computers and purchase VPNs. That kind of cost is very expensive, and our current income cannot afford it.

I still don't think there is a limit over hundreds of accounts. if an error happens while managing hundreds of them please provide it.

As for proxies they aren't currently supported since last I checked it wasn't possible to use them in browsers.

**Comment by benjamin-2019** [comment]
> > > why do you actually want to use proxies? telegram has no IP limit
> > 
> > 
> > Sir, there is no limit for a single account and a single client. When we need to manage hundreds or thousands of accounts on a computer at the same time, this will be limited. It needs dozens or hundreds of proxy IP support. We Now we provide customers with automatic group management services. We can only manage dozens of accounts on a computer. Without proxy IP, we can only add computers and purchase VPNs. That kind of cost is very expensive, and our current income cannot afford it.
> 
> I still don't think there is a limit over hundreds of accounts. if an error happens while managing hundreds of them please provide it.
> 
> As for proxies they aren't currently supported since last I checked it wasn't possible to use them in browsers.

Sir, that’s because you haven’t tried to register hundreds or thousands of accounts on a computer, that requires a lot of mobile phone numbers, you register more than 50-100 mobile phone numbers on a computer, and all subsequent mobile phone numbers are Can no longer receive SMS messages, telegram will no longer send, including your registration in your country with a mobile phone number in another country. I will use the python or php library to implement this part of multi-account login and registration. If this library can be implemented, it will be better. We promise that after we make a profit, we will thank you for your time and contribution.

**Comment by benjamin-2019** [comment]
> > > why do you actually want to use proxies? telegram has no IP limit
> > 
> > 
> > Sir, there is no limit for a single account and a single client. When we need to manage hundreds or thousands of accounts on a computer at the same time, this will be limited. It needs dozens or hundreds of proxy IP support. We Now we provide customers with automatic group management services. We can only manage dozens of accounts on a computer. Without proxy IP, we can only add computers and purchase VPNs. That kind of cost is very expensive, and our current income cannot afford it.
> 
> I still don't think there is a limit over hundreds of accounts. if an error happens while managing hundreds of them please provide it.
> 
> As for proxies they aren't currently supported since last I checked it wasn't possible to use them in browsers.

If the socks5 proxy is currently supported on the command-line side, but the browser side is not supported, it’s not bad, so we don’t have to use two libraries to complete the work of multi-account registration.

**Comment by DricHao** [comment]
YES, it's a problem when you manage many accounts with same IP, Telegram official will ban your APPID. So we need proxy to supporting 'one account one IP'. I'm using gramjs in Node environment.

**Comment by spb-web** [comment]
+

**Comment by iamnnort** [comment]
The same I really need the proxy support for nodejs version of the library

**Comment by 594904020** [comment]
You can modify the source code to support the proxy，I have solved the problem

**Comment by D36u99er** [comment]
> You can modify the source code to support the proxy，I have solved the problem
my telegram is @sandy3306  ,can i get help from you friend. much thanks

**Comment by DricHao** [comment]
> You can modify the source code to support the proxy，I have solved the problem

I've tried to use node-socks to solve this proxy issue but failed, could you share your code bro.

**Comment by 594904020** [comment]
> > You can modify the source code to support the proxy，I have solved the problem
> 
> I've tried to use node-socks to solve this proxy issue but failed, could you share your code bro.

https://www.npmjs.com/package/socket-client-proxy   

in file  “PromisedNetSockets.js”

const Socket = require('socket-client-proxy');

 this.client = new Socket({
    proxy: {
        host: xxxx,
        port: xxxx,
        type: 'socks5', //  socks5  or  http
    },
});

**Comment by painor** [comment]
or use https://gram.js.org/getting-started/authorization#using-mtproxies-and-socks5-proxies

**Comment by 594904020** [comment]
> or use https://gram.js.org/getting-started/authorization#using-mtproxies-and-socks5-proxies

This seems to be unavailable

**Comment by painor** [comment]
> > or use https://gram.js.org/getting-started/authorization#using-mtproxies-and-socks5-proxies
> 
> This seems to be unavailable

what do you mean by unavailable ?
![image](https://user-images.githubusercontent.com/30215770/135439685-ed3ba8fd-d1da-4efc-846d-c185d19001da.png)
